app/src/main/java/org/projectbuendia/client/sync/SyncAdapter.java

```diff
@@ -103,14 +69,23 @@
      * Select a phase by setting a boolean value of true for the appropriate key.
      */
     public enum SyncPhase {
-        SYNC_USERS,
-        SYNC_LOCATIONS,
-        SYNC_CHART_ITEMS,
-        SYNC_CONCEPTS,
-        SYNC_PATIENTS,
-        SYNC_OBSERVATIONS,
-        SYNC_ORDERS,
-        SYNC_FORMS
+        SYNC_USERS(R.string.syncing_users, new UsersSyncPhaseController()),
```

Comment:
I think the phase name and message are handled better at the enum level. By keeping `SyncPhaseController` as a purely functional object it makes it easier to composite logic later.

So my planned next step in this direction is something like have an `IncrementalSyncPhaseController<T> implements SyncPhaseController` that can be used to wrap another function which allows us to generalize incremental sync logic. I think that's cleaner than the approach we're currently using, which is to have helper method calls at the start and end of the `sync()` function for those `SyncPhaseControllers` that do incremental sync.